From 230082c26b377f802e43da277ddf96d204398417 Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Mon, 1 Sep 2014 13:33:14 -0400
Subject: [crypto] qat: use min_t macro

Message-id: <1409578401-26943-23-git-send-email-nikolay@redhat.com>
Patchwork-id: 90099
O-Subject: [RHEL7.1 PATCH BZ 1091130 22/29] crypto: qat - use min_t macro
Bugzilla: 1091130
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Herbert Xu <herbert.xu@redhat.com>

From: Tadeusz Struk <tadeusz.struk@intel.com>

prefer min_t() macro over two open-coded logical tests

Signed-off-by: Bruce Allan <bruce.w.allan@intel.com>
Signed-off-by: Tadeusz Struk <tadeusz.struk@intel.com>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
(cherry picked from commit df0088f50712f9540769c542d0b4d27acc4fc527)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/crypto/qat/qat_common/qat_uclo.c b/drivers/crypto/qat/qat_common/qat_uclo.c
index 17a9954..b1a16eb 100644
--- a/drivers/crypto/qat/qat_common/qat_uclo.c
+++ b/drivers/crypto/qat/qat_common/qat_uclo.c
@@ -901,8 +901,8 @@ static int qat_uclo_set_ae_mode(struct icp_qat_fw_loader_handle *handle)
          (unsigned long *)&handle->hal_handle->ae_mask))
    continue;
   ae_data = &obj_handle->ae_data[ae];
-  for (s = 0; s < ae_data->slice_num && s < ICP_QAT_UCLO_MAX_CTX;
-       s++) {
+  for (s = 0; s < min_t(unsigned int, ae_data->slice_num,
+          ICP_QAT_UCLO_MAX_CTX); s++) {
    if (!obj_handle->ae_data[ae].ae_slices[s].encap_image)
     continue;
    uof_image = ae_data->ae_slices[s].encap_image->img_ptr;
-- 
1.7.1